Since version 1.27.1.685 the update check fails on parameter 'requirements_update_strategy'.

**Describe the bug**
Since version 1.27.1.685 (monday 18-03-2024) the update check (_Checking if FluentValidation 11.5.2 needs updating_) fails on parameter 'requirements_update_strategy'. Expected type T.nilable(Dependabot::RequirementsUpdateStrategy), got type Symbol with value :auto (TypeError).

![image](https://github.com/tinglesoftware/dependabot-azure-devops/assets/60221009/f6ddbea1-d529-4a01-9c8b-3f2b30ede0fd)

`2024-03-20T04:03:45.8363653Z Checking if FluentValidation 11.5.2 needs updating
2024-03-20T04:03:45.8383968Z /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/configuration.rb:296:in 'call_validation_error_handler_default': Parameter 'requirements_update_strategy': Expected type T.nilable(Dependabot::RequirementsUpdateStrategy), got type Symbol with value :auto (TypeError)
2024-03-20T04:03:45.8384728Z Caller: /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/private/methods/call_validation.rb:215
2024-03-20T04:03:45.8385236Z Definition: /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/dependabot-common-0.247.0/lib/dependabot/update_checkers/base.rb:62
2024-03-20T04:03:45.8385775Z 	from /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/configuration.rb:303:in 'call_validation_error_handler'
2024-03-20T04:03:45.8386328Z 	from /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/private/methods/call_validation.rb:300:in 'report_error'
2024-03-20T04:03:45.8386902Z 	from /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/private/methods/call_validation.rb:218:in 'block in validate_call'
2024-03-20T04:03:45.8387459Z 	from /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/private/methods/signature.rb:234:in 'block in each_args_value_type'
2024-03-20T04:03:45.8387999Z 	from /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/private/methods/signature.rb:228:in 'each'
2024-03-20T04:03:45.8388535Z 	from /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/private/methods/signature.rb:228:in 'each_args_value_type'
2024-03-20T04:03:45.8389075Z 	from /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/private/methods/call_validation.rb:215:in 'validate_call'
2024-03-20T04:03:45.8389633Z 	from /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/sorbet-runtime-0.5.11294/lib/types/private/methods/_methods.rb:277:in 'block in _on_method_added'
2024-03-20T04:03:45.8389904Z 	from bin/update_script.rb:339:in 'new'
2024-03-20T04:03:45.8390116Z 	from bin/update_script.rb:339:in 'update_checker_for'
2024-03-20T04:03:45.8390335Z 	from bin/update_script.rb:559:in 'block in <main>'
2024-03-20T04:03:45.8390524Z 	from bin/update_script.rb:545:in 'each'
2024-03-20T04:03:45.8390723Z 	from bin/update_script.rb:545:in '<main>'
2024-03-20T04:03:46.0301400Z ##[error]The process '/usr/bin/docker' failed with exit code 1
2024-03-20T04:03:46.0324507Z ##[section]Finishing: dependabot`

**To Reproduce**
Steps to reproduce the behavior:
1. Run Azure Devops pipeline with task 'dependabot@1' and the DevOps extension installed:
```
steps:
  - task: dependabot@1
    inputs:
      azureDevOpsServiceConnection: <redacted>
```
2. Task will fail on update check

**Expected behavior**
A run without errors

**Screenshots**
See previous screenshot

**Extension (please complete the following information):**
 - Host: Azure DevOps
 - Version 1.27.1.685

**Server (please complete the following information):**
 - Region westeurope
 - Version n.a.

**Additional context**
It seems that this problem is related to an update of Depandabot core [link](https://github.com/dependabot/dependabot-core/pull/9179)

> > Unfortunately, the latest version is still failing with the following error:
> > ```
> > 
> > Status: Downloaded newer image for ghcr.io/tinglesoftware/dependabot-updater-nuget:1.29
> > 
> > warning: parser/current is loading parser/ruby33, which recognizes 3.3.2-compliant syntax, but you are running 3.3.1.
> > 
> > Please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
> > 
> > bin/update_script.rb:180:in `<main>': uninitialized constant RequirementsUpdateStrategy (NameError)
> > 
> > 
> > 
> >     "lockfile-only" => RequirementsUpdateStrategy::LockfileOnly,
> > 
> >                        ^^^^^^^^^^^^^^^^^^^^^^^^^^
> > ```
> 
> Fixed in [1.29.1](https://github.com/tinglesoftware/dependabot-azure-devops/releases/tag/1.29.1)?

Nope, unfortunately not. Firstly, the Azure DevOps task will pull version 1.29 by default (major.minor). And even when explicitly specifying version 1.29.1, it's still broken (task version 1.29.737):

```
Status: Downloaded newer image for ghcr.io/tinglesoftware/dependabot-updater-nuget:1.29.1
warning: parser/current is loading parser/ruby33, which recognizes 3.3.2-compliant syntax, but you are running 3.3.1.
Please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
bin/update_script.rb:181:in `<main>': uninitialized constant RequirementsUpdateStrategy (NameError)

    "lockfile-only" => RequirementsUpdateStrategy::LockfileOnly,
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^
##[error]The process '/usr/bin/docker' failed with exit code 1
```

--- a/updater/bin/update_script.rb
+++ b/updater/bin/update_script.rb
@@ -178,10 +178,10 @@ end
 unless ENV["DEPENDABOT_VERSIONING_STRATEGY"].to_s.strip.empty?
   # [Hash<String, Symbol>]
   VERSIONING_STRATEGIES = {
-    "lockfile-only" => RequirementsUpdateStrategy::LockfileOnly,
-    "widen" => RequirementsUpdateStrategy::WidenRanges,
-    "increase" => RequirementsUpdateStrategy::BumpVersions,
-    "increase-if-necessary" => RequirementsUpdateStrategy::BumpVersionsIfNecessary
+    "lockfile-only" => Dependabot::RequirementsUpdateStrategy::LockfileOnly,
+    "widen" => Dependabot::RequirementsUpdateStrategy::WidenRanges,
+    "increase" => Dependabot::RequirementsUpdateStrategy::BumpVersions,
+    "increase-if-necessary" => Dependabot::RequirementsUpdateStrategy::BumpVersionsIfNecessary
   }.freeze
   strategy_raw = ENV.fetch("DEPENDABOT_VERSIONING_STRATEGY", nil)
   $options[:requirements_update_strategy] = case strategy_raw
@@ -195,8 +195,8 @@ unless ENV["DEPENDABOT_VERSIONING_STRATEGY"].to_s.strip.empty?
   # https://github.com/dependabot/dependabot-core/blob/5926b243b2875ad0d8c0a52c09210c4f5f274c5e/composer/lib/dependabot/composer/update_checker/requirements_updater.rb#L23-L24
   if $package_manager == "npm_and_yarn" || $package_manager == "composer"
     strategy = $options[:requirements_update_strategy]
-    if strategy.nil? || strategy == RequirementsUpdateStrategy::LockfileOnly
-      $options[:requirements_update_strategy] = RequirementsUpdateStrategy::BumpVersions
+    if strategy.nil? || strategy == Dependabot::RequirementsUpdateStrategy::LockfileOnly
+      $options[:requirements_update_strategy] = Dependabot::RequirementsUpdateStrategy::BumpVersions
     end
   end
 
@@ -204,8 +204,8 @@ unless ENV["DEPENDABOT_VERSIONING_STRATEGY"].to_s.strip.empty?
   # https://github.com/dependabot/dependabot-core/blob/ca9f236591ba49fa6e2a8d5f06e538614033a628/pub/lib/dependabot/pub/update_checker.rb#L110
   if $package_manager == "pub"
     strategy = $options[:requirements_update_strategy]
-    if strategy == RequirementsUpdateStrategy::LockfileOnly
-      $options[:requirements_update_strategy] = RequirementsUpdateStrategy::BumpVersions
+    if strategy == Dependabot::RequirementsUpdateStrategy::LockfileOnly
+      $options[:requirements_update_strategy] = Dependabot::RequirementsUpdateStrategy::BumpVersions
     end
   end
 end

